libs/wingc/src/type_check.rs

```diff
@@ -2584,6 +2584,36 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	/// Validate that the given type is a subtype (or same) as the expected type, disregarding whether
+	/// the object is mutable or immutable. If not, add an error to the diagnostics.
+	///
+	/// Returns the given type on success, otherwise returns the expected type.
+	fn validate_equal_types(&mut self, actual_type: TypeRef, expected_type: TypeRef, span: &impl Spanned) -> TypeRef {
```

Comment:
This function name still feels off, it sounds like it's checking if types are equal. Maybe something like this, and we can make a pattern out of `validate_type_*` like we do with `type_check_*`:

```suggestion
	fn validate_type_binary_equality(&mut self, actual_type: TypeRef, expected_type: TypeRef, span: &impl Spanned) -> TypeRef {
```

idk, I'm also struggling to come up with a name...